File: leetcode-submissions/1957_Delete_Characters_to_Make_Fancy_String.cpp
class Solution {
public:
    string makeFancyString(string s) {
        // we need to iterate 3 steps back to the last character
      
        for(int i=0;i<s.size()-3;i++){
            while(s[i]==s[i+1] and s[i+1]==s[i+2]){
              s.erase(i,1); 
            }
        }
        return s;
    }
};

File: leetcode-submissions/1957_Delete_Characters_to_Make_Fancy_String.ts
function makeFancyString(s: string): string {
    const st:string[]=[];
    for(let i:number=0 ;i<s.length;i++){
        if(st.length>=2 && st[st.length-1]===s[i] && st[st.length-2]===s[i]){
            continue;
        }
        st.push(s[i]);
    }
    return st.join('');
};